### PR TITLE
IS-912: Implement copy command

### PR DESCRIPTION
**Pull request by boyeon555**

### CODE REVIEW

File: icondbtools/command/command_copy.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+        block_reader = BlockDatabaseReader()
+        block_reader.open(db_path)
+
+        new_db = plyvel.DB(new_db_path, create_if_missing=True)
+
+        with new_db.write_batch() as wb:
+            last_block_hash = None
+            for height in range(start, start + count):
+                block_height_key = block_reader.get_block_height_key(height)
```
Comment (goldworm-icon) [inline icondbtools/command/command_copy.py]: Add a type hint for block_height_key

File: icondbtools/libs/block_database_raw_reader.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    def get_block_by_height(self, block_height: int) -> bytes:
+        """Get block in bytes by block height     
+
+        :param block_height: block height in integer
+        :return: block in bytes
+        """
+        block_height_key: bytes = self.get_block_height_key(block_height)
+        block_hash: bytes = self._db.get(block_height_key)
+        if block_hash is None:
+            return
```
Comment (goldworm-icon) [inline icondbtools/libs/block_database_raw_reader.py:38]: Fix type hint mismatch warning

File: icondbtools/libs/block_database_raw_reader.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        """Get last block in bytes
+
+        :return: block in bytes
+        """
+        last_block_key = b'last_block_key'
+        block_hash: bytes = self._db.get(last_block_key)
+        block: bytes = self._db.get(block_hash)
+        return block
+
+    def get_block_by_height(self, block_height: int) -> bytes:
```
Comment (goldworm-icon) [inline icondbtools/libs/block_database_raw_reader.py]: Where is get_block_by_hash()?

**Review by goldworm-icon — CHANGES_REQUESTED** [review]

File: icondbtools/command/command_copy.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+        with new_db.write_batch() as wb:
+            for height in range(start, start + count):
+                block_height_key: bytes = block_reader.get_block_height_key(height)
+                block: bytes = block_reader.get_block_by_height(height)
+
+                if block is None:
+                    break
+
+                block_hash: bytes = json.loads(block)["block_hash"].encode()
```
Comment (goldworm-icon) [inline icondbtools/command/command_copy.py]: Find the other way to get block_hash without block data parsing

**Review by goldworm-icon — CHANGES_REQUESTED** [review]

File: icondbtools/command/command_copy.py (hunk 1 above)
Comment (boyeon555) [reply icondbtools/command/command_copy.py]: Added

File: icondbtools/libs/block_database_raw_reader.py (hunk 2 above)
Comment (boyeon555) [reply icondbtools/libs/block_database_raw_reader.py:38]: Fixed

File: icondbtools/command/command_copy.py (hunk 4 above)
Comment (boyeon555) [reply icondbtools/command/command_copy.py]: Added 'get_value_by_height' method for it.

File: icondbtools/libs/block_database_raw_reader.py (hunk 3 above)
Comment (boyeon555) [reply icondbtools/libs/block_database_raw_reader.py]: Added

**Review by goldworm-icon — APPROVED** [review]

**Pull request merged**